rmw_fastrtps_shared_cpp/src/rmw_take.cpp

```diff
@@ -110,26 +133,45 @@ _take_sequence(
   CustomSubscriberInfo * info = static_cast<CustomSubscriberInfo *>(subscription->data);
   RCUTILS_CHECK_FOR_NULL_WITH_MSG(info, "custom subscriber info is null", return RMW_RET_ERROR);
 
```

Comment:
`rmw_cyclonedds` checks that the `count` provided by the user is not greater than the limit allowed by their DDS implementation:

```
  if (count > (std::numeric_limits<uint32_t>::max)()) {
    RMW_SET_ERROR_MSG_WITH_FORMAT_STRING(
      "Cannot take %zu samples at once, limit is %d",
      count, (std::numeric_limits<uint32_t>::max)());
    return RMW_RET_ERROR;
  }
```

Fast DDS uses instead of `uint32_t`, `int32_t` because `LENGTH_UNLIMITED` corresponds to value -1.

This check should be added in `__rmw_take_sequence` before calling `_take_sequence` (around L269)

Reply:
N/A